File: test_rosbag/src/RandomDataPublisher.cpp
#include <ros/ros.h>
#include <test_rosbag/Int16Stamped.h>   // The custom made message
#include <stdio.h>  // For rand() function


int main(int argc, char **argv) {
    // Initialize the node
    ros::init(argc, argv, "CPP_RandomNumberGenerator");
    // Set up the node handler
    ros::NodeHandle node_handler;
    // Create publisher object
    ros::Publisher rnum_publisher = node_handler.advertise<test_rosbag::Int16Stamped>("rnumber", 10);
    // Create rate of publishing
    ros::Rate rate_handler(2); // 2 Hz
    // Sequence
    int i = 0;
    while(ros::ok()) {  // While ROS is working
        // Message to publish
        test_rosbag::Int16Stamped msg;
        msg.data = rand();  // Data
        msg.header.seq = ++i;   // Sequence (with increment)
        msg.header.stamp = ros::Time::now();    // Time
        msg.header.frame_id = "/map";   // Frame of the message
        // Publish the message
        rnum_publisher.publish(msg);
        ROS_INFO("Value: %d", msg.data);
        // Delay for adjusting to the rate
        rate_handler.sleep();
    }
}

File: test_rosbag/scripts/RandomDataPublisher.py
#!/usr/bin/env python
import rospy
from test_rosbag.msg import Int16Stamped
from std_msgs.msg import Int16
import random

def main():
    # Initialize node
    rospy.init_node("PY_RandomNumberGenerator")
    # Create publisher
    rnumber_publisher = rospy.Publisher("rnumber", Int16Stamped, queue_size=50)
    # Rate of publishing
    rate_controller = rospy.Rate(2)
    i = 0   # Sequence
    while(not rospy.is_shutdown()):
        # Create message
        i = i+1
        msg = Int16Stamped()
        # Assign data
        msg.data = random.randint(-32767, 32768)
        msg.header.seq = i
        msg.header.stamp = rospy.Time.now()
        msg.header.frame_id = "/map"
        # Publish message
        rnumber_publisher.publish(msg)
        rospy.loginfo("Value = %d", msg.data)
        # Delay
        rate_controller.sleep()


if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("Keyboard interrupt detected")
